ref/general/application-hardening.adoc

```diff
@@ -61,25 +65,50 @@ To alleviate this risk, use wildcards to secure servlets wherever possible.
 If that is not appropriate, carefully double check your `web.xml` file before deployment.
 
 When you specify authorization constraints for servlets, ensure that you either list no methods or carefully list all the methods for a servlet, including GET, POST, PUT, HEAD, etc.
-It's likely that you must list all the methods if you are using multiple constraints.
+If you are using multiple constraints, list all the methods.
 
-In the following secure example, endpoints in the web application are unreachable by default.
-If a new servlet is added, you must add a security constraint for the new URL so that anyone can use it:
+Specify the `<deny-uncovered-http-methods>` element] in the `web.xml` file.
+When the `<deny-uncovered-http-methods>` element is specified, the container denies any uncovered HTTP methods that aren't listed in the security constraints for a URL pattern that best matches the request URL.
+A `403 (SC_FORBIDDEN)` status code is returned by the denied HTTP methods.
+You can also use the link:https://openliberty.io/docs/ref/config/#webApplication.html#webservices-bnd/http-publishing/webservice-security/security-constraint/web-resource-collection&expand=true[`<http-method-omission>` element] to specify an HTTP method to which a security constraint shouldn't apply.
+In the following secure example, the `<deny-uncovered-http-methods>` element is specified in `web.xml` file:
 
 [source,xml]
 ----
-<security-constraint>
-  <web-resource-collection>
-    <web-resource-name>DefaultDeny</web-resource-name>
-    <url-pattern>/*</url-pattern>
-  </web-resource-collection>
-  <auth-constraint>
-    <role-name>NoAccess</role-name>
-  </auth-constraint>
+<servlet-mapping id="ServletMapping_1">
+		<servlet-name>MyServlet</servlet-name>
+		<url-pattern>/MyURLPattern</url-pattern>
+</servlet-mapping>
+
+<deny-uncovered-http-methods/>
+
+<!-- SECURITY CONSTRAINTS -->
+<security-constraint id="SecurityConstraint_1">
+		<web-resource-collection id="WebResourceCollection_1">
+			<web-resource-name>Protected with Employee or Manager roles</web-resource-name>
+			<url-pattern>/MyURLPattern</url-pattern>
+			<http-method>GET</http-method>
+			<http-method>POST</http-method>
+		</web-resource-collection>
+		<auth-constraint id="AuthConstraint_1">
+			<role-name>Employee</role-name>
+			<role-name>Manager</role-name>
+		</auth-constraint>
 </security-constraint>
 ----
 
-For link:/docs/ref/feature/#servlet-3.1.html[Servlet 3.1] or higher, use the `<deny-uncovered-http-methods/>` or `<http-method-omission>` elements.
+When the `<deny-uncovered-http-methods>` element is specified in the `web.xml` file, a message is also logged in the `messages.log` file for each URL pattern in each servlet. This message indicates the uncovered methods with a note that those methods are unprotected and not accessible, as shown in the following example:
```

Comment:
a message is also logged in the `messages.log` file

not sure you need "also" here

This message indicates the uncovered methods

should this be plural? The previous sentence says "a message is also logged in the `messages.log` file for each URL pattern in each servlet", implying more than one message- unless it's the same message every time- which maybe it is.

Reply:
I took out "also". Yeah, I think you're right that it should be plural since there's a message logged for each URL pattern in each servlet. Update.